### PR TITLE
Refactored Core/Category unit tests so they inherit from AbstractServer

### PR DESCRIPTION
**Pull request by devos50**

### CODE REVIEW

**Comment by devos50** [comment]
Okay, this is fun to notice: we are not stopping the i2i server. Guess this extra check is very useful to have :+1:

**Comment by whirm** [comment]
@devos50 ready?

**Comment by devos50** [comment]
@whirm not yet, tests are hanging :/

File: Tribler/Test/test_as_server.py [hunk 1]
```diff
@@ -124,6 +125,11 @@ def tearDown(self, annotate=True):
         self.assertFalse(delayed_calls, "The reactor was dirty when tearing down the test")
         self.assertFalse(Session.has_instance(), 'A session instance is still present when tearing down the test')
 
+        # Check whether we have closed all the sockets
+        open_readers = reactor.getReaders()
+        for reader in open_readers:
+            self.assertNotIsInstance(reader, BasePort, "The reactor left an open listening port: %s" % reader)
```
Comment (whirm) [inline Tribler/Test/test_as_server.py]: "The test left a listening port behind"

File: Tribler/Utilities/Instance2Instance.py [hunk 2]
```diff
@@ -29,11 +30,16 @@ class I2IFactory(Factory):
...
-    def __init__(self, i2iport, callback):
-        endpoint = TCP4ServerEndpoint(reactor, i2iport, interface="127.0.0.1")
-        endpoint.listen(I2IFactory(callback))
+    def __init__(self, i2iport):
+        self.endpoint = TCP4ServerEndpoint(reactor, i2iport, interface="127.0.0.1")
+
+    @inlineCallbacks
+    def start(self, callback):
+        self.listenport = yield self.endpoint.listen(I2IFactory(callback))
+        returnValue(self.listenport)
```
Comment (whirm) [inline Tribler/Utilities/Instance2Instance.py]: There's no stop method and you need to update Tribler to start and stop the instance2instance stuff.

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
retest this please

**Comment by devos50** [comment]
@whirm I'm finished with this PR. I don't think it makes sense to hunt coverage on a test that is never executed and a line on the tunnel helper startup script (unless you want a test for the script to start up the tunnel helpers but I don't think that's necessary since the core logic of the tunnel community is not in `Tribler/community/tunnel/main.py`).

**Pull request merged**